### PR TITLE
Support local images for preview

### PR DESCRIPTION
**Pull request by wcandillon**

### CODE REVIEW

**Comment by wcandillon** [comment]
@vkedwardli @Piropa Do you guys like this PR? 🤔

**Comment by vkedwardli** [comment]
No backward compatibility, but the code likes much cleaner now, should be able to satisfy @Piropa's usage now

File: src/Image.js [hunk 1]
```diff
@@ -4,12 +4,13 @@ import * as React from "react";
...
 import {BlurView} from "expo";
 import {type StyleObj} from "react-native/Libraries/StyleSheet/StyleSheetTypes";
+import type {ImageSourcePropType} from "react-native/Libraries/Image/ImageSourcePropType";
 
 import CacheManager from "./CacheManager";
 
 type ImageProps = {
     style?: StyleObj,
-    preview?: string,
+    preview?: ImageSourcePropType,
```
Comment (LouisJS) [inline src/Image.js]: Would that make `preview` as an URL invalid ?

**Comment by LouisJS** [comment]
I'm using `react-native-expo-image-cache": "^2.0.0`.

According to this PR, i should replace my `preview` props from :

```
<Image
      style={style.image}
      {...{
       preview: "https://t3.gstatic.com/images?q=tbn:ANd9GcRYiGRk0xt6OXpRa3j7ySJNeATepTe7zaWzm1G6KpGAwpHePcTmYPYp6WI",
       uri: "http://pic.xeeok.com/uploads/20170821/201708210730182752/60764_012.jpg"
    }}
/>
```

to

```
<Image
      style={style.image}
      {...{
       preview: {uri: "https://t3.gstatic.com/images?q=tbn:ANd9GcRYiGRk0xt6OXpRa3j7ySJNeATepTe7zaWzm1G6KpGAwpHePcTmYPYp6WI"},
       uri: "http://pic.xeeok.com/uploads/20170821/201708210730182752/60764_012.jpg"
    }}
/>
```

Result :

screen shot 2018-03-08 at 14 44 35

In which case should i use the format
```preview: "https://my_url.jpg"```
or 
```preview: {uri: "https://my_url.jpg"}```

**Comment by wcandillon** [comment]
You should use: preview: `{uri: "https://my_url.jpg"}`
It's possible that again it's an issue with the npm package itself 🤔

**Comment by wcandillon** [comment]
@LouisJS should be fixed in `2.0.1` now.

**Pull request merged**